### PR TITLE
Fix(#857) ProxiedComponents Type Checking

### PR DESCRIPTION
**Pull request by liviudobrea**
Closes IBM/carbon-components-react#857

Add internal helper for proxied component type checking.

#### Changelog

**New**

* React-proxy dev dependency

**Changed**

* childrenOf and childrenOf-Type

### CODE REVIEW

File: src/prop-types/tools/areComponentsEqual.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const RHL_PREFIX = '__reactstandin__getCurrent';
+
+const getProxyOrType = c => {
+  const isProxy = c && typeof c !== 'string' && Reflect.has(c, RHL_PREFIX);
```
Comment (asudoh) [inline src/prop-types/tools/areComponentsEqual.js]: `Reflect` is not supported by IE11 - Any alternatives?

File: src/prop-types/tools/areComponentsEqual.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  return isProxy ? c[RHL_PREFIX]() : c;
+};
+
+const areComponentsEqual = (a, b) => {
```
Comment (asudoh) [inline src/prop-types/tools/areComponentsEqual.js]: Probably we'd take an array for `b`, or make this function varargs?

File: src/prop-types/tools/areComponentsEqual.js (hunk 1 above)
Comment (liviudobrea) [reply src/prop-types/tools/areComponentsEqual.js]: Sure, will add fallback.

File: src/prop-types/tools/areComponentsEqual.js (hunk 2 above)
Comment (liviudobrea) [reply src/prop-types/tools/areComponentsEqual.js]: Yep, we can, but I wouldn't do that in the helper method though. I still prefer this https://github.com/IBM/carbon-components-react/blob/47b7447f18792f8d1655863f75dcf749cec19ce4/src/prop-types/childrenOf.js#L20-L33

**Comment by joshblack** [comment]
Also, make sure to run `yarn prettier` so that ci checks pass on formatting 😄

**Comment by liviudobrea** [comment]
There we go, replaced `Reflect` in favour of globally supported js API. Ran `prettier` and tested with our own project successfully.

**Comment by liviudobrea** [comment]
@joshblack Can't merge master inside branch from git UI until
```
Merging is blocked
Merging can be performed automatically once the requested changes are addressed.
```
Doesn't look like it's something that I can do from my side.

**Comment by alisonjoseph** [comment]
Closing this due to inactivity. Feel free to reopen if this is something you still need and have time to make the requested updates.

**Pull request closed without merging**